src/mlmc/sampling_pool.py

```diff
@@ -51,39 +53,133 @@ def compute_seed(sample_id):
     def calculate_sample(sample_id, level_sim):
         """
         Method for calculating results
-        :param sample_id:
-        :param level_sim:
+        :param sample_id: str
+        :param level_sim: LevelSimulation
         :return:
         """
         seed = SamplingPool.compute_seed(sample_id)
         res = (None, None)
         err_msg = ""
+        running_time = 0
         try:
+            start = time.time()
             res = level_sim.calculate(level_sim.config_dict, seed)
+            running_time = time.time() - start
         except Exception as err:
             err_msg = str(err)
 
-        return sample_id, res, err_msg
+        return sample_id, res, err_msg, running_time
+
+    def _change_to_sample_directory(self, path: str):
+        """
+        Create sample directory and change working directory
+        :param path: str
+        :return: None
+        """
+        sample_dir = os.path.join(self._work_dir, path)
+        if not os.path.isdir(sample_dir):
+            os.makedirs(sample_dir, mode=0o775, exist_ok=True)
+        os.chdir(sample_dir)
```

Comment:
Call chdir just before call of the simulation (i.e. in calculate_sample) and then 
change dir back. Consider a context manager:
```
class change_cwd:
    """
    Context manager that change CWD, to given relative or absolute path.
    """
    def __init__(self, path: str):
        self.path = path
        self.orig_cwd = ""

    def __enter__(self):
        if self.path:
            self.orig_cwd = os.getcwd()
            os.chdir(self.path)

    def __exit__(self, exc_type, exc_value, traceback):
        if self.orig_cwd:
            os.chdir(self.orig_cwd)
```

Reply:
Also consider rename of the method.